js/ParticipantProgress.js

```diff
@@ -1,7 +1,135 @@
-function UpdateInfo() {
-    var participantId = GLOBALS.PARTICIPANT_ID;
-    var personalData = GLOBALS.GET_PARTICIPANT_INFO(participantId);
-    //console.log(personalData);
-    var donationData = GLOBALS.GET_PARTICIPANT_DONATION_INFO(participantId);
-    //console.log(donationData);
-}
\ No newline at end of file
+// function UpdateInfo() {
+//     var participantId = GLOBALS.PARTICIPANT_ID;
+//     var personalData = GLOBALS.GET_PARTICIPANT_INFO(participantId);
+//     //console.log(personalData);
+//     var donationData = GLOBALS.GET_PARTICIPANT_DONATION_INFO(participantId);
+//     //console.log(donationData);
+// }
+
+(function() {
+  //copied from http://papermashup.com/read-url-get-variables-withjavascript/
+  function getUrlVars() {
+    var vars = {};
+      var parts = window.location.href.replace(/[?&]+([^=&]+)=([^&]*)/gi, function(m,key,value) {
+        vars[key] = value;
+      });
+    return vars;
+  }
+
+  var app = angular.module( 'progress', [] );
+
+  app.controller( 'progressCtrl', function( $scope, $http ) { 
+  
+    var urlVars = getUrlVars();
+    var uid = urlVars['id'];
+
+    //borrowed from http://axonflux.com/handy-rgb-to-hsl-and-rgb-to-hsv-color-model-c
+    var hslToRgb = function(h, s, l){
```

Comment:
i'm gonna wait and see if anyone else actually needs it before doing that. but i'll keep it in mind